nearcore/src/config.rs

```diff
@@ -589,6 +599,9 @@ impl NearConfig {
                 tracked_shards_config: config.tracked_shards_config(),
                 archive: config.archive,
                 save_trie_changes: config.save_trie_changes.unwrap_or(!config.archive),
+                save_tx_outcomes: config
+                    .save_tx_outcomes
+                    .unwrap_or_else(|| config.archive || config.rpc_addr().is_some()),
```

Comment:
I think there is some ambiguity in our code:

* `rpc_addr` means that node can respond to status queries (e.g. debug ui) which is basically every node
* however, we call "RPC node" a node which tracks all shards and respond to user queries (view account, get tx status).

So "if tracked_shards_config == AllShards" seems a better fit here.